Archipelago.cpp

```diff
@@ -487,22 +487,41 @@ void AP_RegisterSetReplyCallback(void (*f_setreply)(AP_SetReply)) {
     setreplyfunc = f_setreply;
 }
 
-void AP_SetNotify(std::map<std::string,AP_DataType> keylist) {
+void AP_SetNotify(std::map<std::string,AP_DataType> keylist, bool requestCurrentValue) {
     Json::Value req_t;
     req_t[0]["cmd"] = "SetNotify";
+
     int i = 0;
     for (std::pair<std::string,AP_DataType> keytypepair : keylist) {
         req_t[0]["keys"][i] = keytypepair.first;
         map_serverdata_typemanage[keytypepair.first] = keytypepair.second;
+
+        if (requestCurrentValue){
```

Comment:
This should probably be the first operation, not the second.
Additionally, is it possible to instead use SetServerData here? You could define the operations in bulk, then commit before using APSend for the setnotify

Reply:
No it should litterally be last operations, so you are subscribed to the key at that point

Reply:
I can live with both